lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -4,153 +4,615 @@
+
+  /// The key value of the key represented by this event.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key>
   external String get key;
+
+  /// Whether the `Meta` key (on Mac keyboards, the `⌘ Command key`; on Windows keyboards, the Windows key (`⊞`))
+  /// was active when the key event was generated.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/metaKey>
   external bool get metaKey;
+
+  /// Whether the key is being held down such that it is automatically repeating.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat>
   external bool get repeat;
+
+  /// Whether the `Shift` key was active when the this event was generated.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/shiftKey>
   external bool get shiftKey;
+
+  /// A system and implementation dependent numerical code identifying the unmodified
+  /// value of the pressed key.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode>
   external num get keyCode;
+
+  /// The Unicode reference number of the key; this attribute is used only by the `keypress` event.
+  ///
+  /// For keys whose `char` attribute contains multiple characters, this is the Unicode value of the first character in
+  /// that attribute. In Firefox 26 this returns codes for printable characters.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode>
   external num get charCode;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [CompositionEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/CompositionEvent>
 @JS()
-@anonymous
 class SyntheticCompositionEvent extends SyntheticEvent {
+  /// [SyntheticCompositionEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticCompositionEvent` instead.
+  external factory SyntheticCompositionEvent._();
+
+  /// The characters generated by the input method that raised this; its varies depending on the type of event
+  /// that generated the this object.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/CompositionEvent/data>
   external String get data;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [FocusEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent>
 @JS()
-@anonymous
 class SyntheticFocusEvent extends SyntheticEvent {
-  external EventTarget get relatedTarget;
+  /// [SyntheticFocusEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticFocusEvent` instead.
+  external factory SyntheticFocusEvent._();
+
+  /// A secondary target for this event.
+  ///
+  /// In some cases (such as when tabbing in or out a page), this property may be set to `null` for security reasons.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent/relatedTarget>
+  external dynamic get relatedTarget;
```

Comment:
In that same [comment from Greg](https://github.com/cleandart/react-dart/pull/284#issuecomment-729270295), he noted that `relatedTarget` needs to be `dynamic` because consumers make calls to `EventTarget` subclasses